### DISCUSSION TITLE
MQTT SSL/TLS: (E) Failed to connect to MQTT: -9

### DISCUSSION THREAD

**Discussion by MichaelUray** [UtilitechAS/amsreader-firmware#621] [Q&A]
I receive the above mentioned error when I try to connect to a MQTT broker/server via SSL.

The debug via telnet in verbose mode shows the following messages:
```
(W) MQTT connection not available, double blink
(D) Received price API config
(I) Saving configuration now...
(I) Successfully saved.
(D) Serving /data.json over http...
(D) Disconnecting MQTT before connecting
(I) MQTT SSL is configured (230676kb free heap)
(I) Connecting to MQTT mqtt.xxx.com:8883
(E) Failed to connect to MQTT: -9
```

I have uploaded the root certificate in DER format and it shows it is ok:
![2023-08-21_08-56-34_AMS_reader_–_Mozilla_Firefox](https://github.com/UtilitechAS/amsreader-firmware/assets/25169478/1e808566-0691-4f6b-9fdd-70b383550cfa)

What does the error message `(E) Failed to connect to MQTT: -9`mean and what could be wrong?

**Comment by gskjold** [comment]
The way MQTT/SSL is implemented in this firmware, it will require you to provide a client cert and key. There might be a way around that, but I am not too familiar with MQTT/SSL to tell.

Error code -9 have description "Missing og wrong packet"
